src/EnergyPlus/DataSurfaces.cc

```diff
@@ -498,6 +498,215 @@ namespace DataSurfaces {
+
+			// Using/Aliasing
+			using DataEnvironment::EarthRadius;
+			using DataEnvironment::SiteTempGradient;
+			using DataEnvironment::WeatherFileTempModCoeff;
+
+			if ( SiteTempGradient == 0.0 ) {
+				OutDryBulbTemp = DataEnvironment::OutDryBulbTemp;
+				OutWetBulbTemp = DataEnvironment::OutWetBulbTemp;
+			} else {
+				// Base temperatures at Z = 0 (C)
+				Real64 const BaseDryTemp( DataEnvironment::OutDryBulbTemp + WeatherFileTempModCoeff );
+				Real64 const BaseWetTemp( DataEnvironment::OutWetBulbTemp + WeatherFileTempModCoeff );
+
+				Real64 const Z( Centroid.z ); // Centroid value
+				if ( Z <= 0.0 ) {
+					OutDryBulbTemp = BaseDryTemp;
+					OutWetBulbTemp = BaseWetTemp;
+				} else {
+					OutDryBulbTemp = BaseDryTemp - SiteTempGradient * EarthRadius * Z / ( EarthRadius + Z );
+					OutWetBulbTemp = BaseWetTemp - SiteTempGradient * EarthRadius * Z / ( EarthRadius + Z );
+				}
+			}
+		}
+
+		void
+		SurfaceData::
+		SetWindSpeedAt( Real64 const fac )
+		{
+			// SUBROUTINE INFORMATION:
+			//       AUTHOR         Linda Lawrie
+			//       DATE WRITTEN   June 2013
+			//       MODIFIED       na
+			//       RE-ENGINEERED  na
+
+			// PURPOSE OF THIS SUBROUTINE:
+			// Routine provides facility for doing bulk Set Windspeed at Height.
+
+			// Using/Aliasing
+			using DataEnvironment::SiteWindExp;
+
+			if ( SiteWindExp == 0.0 ) {
+				WindSpeed = DataEnvironment::WindSpeed;
+			} else {
+				Real64 const Z( Centroid.z ); // Centroid value
+				if ( Z <= 0.0 ) {
+					WindSpeed = 0.0;
+				} else {
+					//  [Met] - at meterological Station, Height of measurement is usually 10m above ground
+					//  LocalWindSpeed = Windspeed [Met] * (Wind Boundary LayerThickness [Met]/Height [Met])**Wind Exponent[Met] &
+					//                     * (Height above ground / Site Wind Boundary Layer Thickness) ** Site Wind Exponent
+					WindSpeed = fac * std::pow( Z, SiteWindExp );
+				}
+			}
+		}
+
+		// Computed Shape Category
+		ShapeCat
+		SurfaceData::
+		computed_shapeCat() const
+		{
+			if ( Shape == Triangle ) {
+				return ShapeCat::Triangular;
+			} else if ( Shape == TriangularWindow ) {
+				return ShapeCat::Triangular;
+			} else if ( Shape == TriangularDoor ) {
+				return ShapeCat::Triangular;
+			} else if ( Shape == Rectangle ) {
+				return ShapeCat::Rectangular;
+			} else if ( Shape == RectangularDoorWindow ) {
+				return ShapeCat::Rectangular;
+			} else if ( Shape == RectangularOverhang ) {
+				return ShapeCat::Rectangular;
+			} else if ( Shape == RectangularLeftFin ) {
+				return ShapeCat::Rectangular;
+			} else if ( Shape == RectangularRightFin ) {
+				return ShapeCat::Rectangular;
+			} else if ( IsConvex ) {
+				return ShapeCat::Convex;
+			} else {
+				return ShapeCat::Nonconvex;
+			}
+		}
+
+		// Computed Plane
+		SurfaceData::Plane
+		SurfaceData::
+		computed_plane() const
+		{
+			Vertices::size_type const n( Vertex.size() );
+			assert( n >= 3 );
```

Comment:
@DeadParrot This is `RefBldgOutPatientNew2004_Chicago.idf` in the ExampleFiles distributed with EnergyPlus.